packages/commonwealth/client/scripts/views/pages/view_snapshot_proposal/index.tsx

```diff
@@ -13,32 +10,26 @@ import {
 } from 'helpers/snapshot_utils';
 import useNecessaryEffect from 'hooks/useNecessaryEffect';
 import { LinkSource } from 'models/Thread';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
 import app from 'state';
-import { useGetThreadsByLinkQuery } from 'state/api/threads';
+import { notifyError } from '../../../controllers/app/notifications';
+import useManageDocumentTitle from '../../../hooks/useManageDocumentTitle';
 import AddressInfo from '../../../models/AddressInfo';
-import { CWContentPage } from '../../components/component_kit/CWContentPage';
+import { useGetThreadsByLinkQuery } from '../../../state/api/threads/index';
+import { CWContentPage } from '../../components/component_kit/CWContentPage/index';
 import {
   ActiveProposalPill,
   ClosedProposalPill,
 } from '../../components/proposal_pills';
 import { QuillRenderer } from '../../components/react_quill_editor/quill_renderer';
+import { PageLoading } from '../loading';
 import { SnapshotInformationCard } from './snapshot_information_card';
 import { SnapshotPollCardContainer } from './snapshot_poll_card_container';
 import { SnapshotVotesTable } from './snapshot_votes_table';
-import useBrowserWindow from 'hooks/useBrowserWindow';
-import useManageDocumentTitle from '../../../hooks/useManageDocumentTitle';
 
-type ViewSnapshotProposalPageProps = {
-  identifier: string;
-  scope: string;
-  snapshotId: string;
-};
-
-export const ViewSnapshotProposalPage = ({
-  identifier,
-  snapshotId,
-}: ViewSnapshotProposalPageProps) => {
-  const [proposal, setProposal] = useState<SnapshotProposal | null>(null);
+export function useSnapshotProposalData(snapshotProposalId, snapshotId) {
```

Comment:
- hooks should not be in component files. move this to `useSnapshotProposalData.tsx` . This will declutter this component a lot. Furthemore, you are linking two components as a dependencies because of the hook import
- interface UseSnapshotProposalDataProps is missing